pkg/azuredisk/azuredisk_v2.go

```diff
@@ -110,60 +114,121 @@ type DriverV2 struct {
 	azDriverNodeInformer              azdiskinformertypes.AzDriverNodeInformer
 	deviceChecker                     *deviceChecker
 	kubeClientQPS                     int
+	isControllerPlugin                bool
+	isNodePlugin                      bool
 }
 
 // NewDriver creates a driver object.
-func NewDriver(options *DriverOptions) CSIDriver {
-	return newDriverV2(options, *driverObjectNamespace, *nodePartition, *controllerPartition, *heartbeatFrequencyInSec, *controllerLeaseDurationInSec, *controllerLeaseRenewDeadlineInSec, *controllerLeaseRetryPeriodInSec, *leaderElectionNamespace)
+func NewDriver(config *azdiskv1beta2.AzDiskDriverConfiguration) CSIDriver {
+	getDriverConfig(config)
+	return newDriverV2(config)
+}
+
+func getDriverConfig(driverConfig *azdiskv1beta2.AzDiskDriverConfiguration) {
+	if *driverConfigPath != "" {
+		if driverConfig.ObjectNamespace == "" {
+			driverConfig.ObjectNamespace = consts.DefaultAzureDiskCrdNamespace
+		}
+		if driverConfig.ControllerConfig.Enabled == nil {
```

Comment:
I performed an experiment to look at the behavior of `yaml.Unmarshal` on field values that are not included in the yaml stream. It appears that it does not overwrite them with a default value if the field is absent from the stream. Given this behavior, we can avoid the use of pointer fields for optional values. Instead, we simply initialize the `AzDriverConfiguration` with default values before calling `yaml.Unmarshal` to unmarshal the values into it. This approach could simplify this function quite a bit.

Reply:
Changed the code accordingly.